osquery/tables/utility/file.cpp

```diff
@@ -99,6 +99,34 @@ void genFileInfo(const fs::path& path,
     r["type"] = "unknown";
   }
 
+#else
+
+  WINDOWS_STAT file_stat;
+
+  if (!platformStat(path, &file_stat).ok()) {
+    return;
```

Comment:
Ah, I think this is the issue that's causing Windows failures. Seems like the `stat` you've got isn't working during test runs, seems like a separate problem that we should sort out, however you'll want to `results.push_back(r);` here before returning, to ensure that we at least get some data back. This is passing test runs for me. Again, we should sort out what's up with the `platformStat` failing here. Perhaps add a `VLOG(1) << "Platform Stat failed for " << path;`?

Reply:
Looks like there's another test failure after this one, I'll take a look when I get a minute today, if not I should be able to get to it this weekend.

Reply:
After adding some error checking and your recommendations I'm getting things like this:

```
osquery> select * from file where directory="C:\Windows";
I0706 15:16:57.491475  9800 file.cpp:108] Platform Stat failed for C:\Windows\addins with Access is denied.
I0706 15:16:57.491475  9800 file.cpp:108] Platform Stat failed for C:\Windows\appcompat with Access is denied.
I0706 15:16:57.492513  9800 file.cpp:108] Platform Stat failed for C:\Windows\apppatch with Access is denied.
I0706 15:16:57.492513  9800 file.cpp:108] Platform Stat failed for C:\Windows\AppReadiness with Access is denied.
I0706 15:16:57.492513  9800 file.cpp:108] Platform Stat failed for C:\Windows\assembly with Access is denied.
I0706 15:16:57.492513  9800 file.cpp:108] Platform Stat failed for C:\Windows\bcastdvr with Access is denied.
I0706 15:16:57.495477  9800 file.cpp:108] Platform Stat failed for C:\Windows\BitLockerDiscoveryVolumeContents with Access is denied.
I0706 15:16:57.495477  9800 file.cpp:108] Platform Stat failed for C:\Windows\Boot with Access is denied.
I0706 15:16:57.496496  9800 file.cpp:108] Platform Stat failed for C:\Windows\bootstat.dat with The process cannot access the file because it is being used by an
other process.
I0706 15:16:57.496496  9800 file.cpp:108] Platform Stat failed for C:\Windows\Branding with Access is denied.
I0706 15:16:57.496496  9800 file.cpp:108] Platform Stat failed for C:\Windows\CbsTemp with Access is denied.
I0706 15:16:57.497475  9800 file.cpp:108] Platform Stat failed for C:\Windows\Containers with Access is denied.
I0706 15:16:57.498508  9800 file.cpp:108] Platform Stat failed for C:\Windows\CSC with Access is denied.
I0706 15:16:57.498508  9800 file.cpp:108] Platform Stat failed for C:\Windows\Cursors with Access is denied.

...
```

and 

```I0706 15:18:04.008680  9800 virtual_table.cpp:301] Error casting inode () to BIGINT
I0706 15:18:04.008680  9800 virtual_table.cpp:301] Error casting uid () to BIGINT
I0706 15:18:04.008680  9800 virtual_table.cpp:301] Error casting gid () to BIGINT
I0706 15:18:04.008680  9800 virtual_table.cpp:301] Error casting device () to BIGINT
I0706 15:18:04.008680  9800 virtual_table.cpp:301] Error casting size () to BIGINT
I0706 15:18:04.009670  9800 virtual_table.cpp:292] Error casting block_size () to INTEGER
I0706 15:18:04.009670  9800 virtual_table.cpp:301] Error casting atime () to BIGINT
I0706 15:18:04.009670  9800 virtual_table.cpp:301] Error casting mtime () to BIGINT
I0706 15:18:04.009670  9800 virtual_table.cpp:301] Error casting ctime () to BIGINT
I0706 15:18:04.009670  9800 virtual_table.cpp:301] Error casting btime () to BIGINT
I0706 15:18:04.010646  9800 virtual_table.cpp:292] Error casting hard_links () to INTEGER

         path = C:\Windows\WinSxS
    directory = C:\Windows
     filename = WinSxS
        inode =
          uid =
          gid =
         mode =
       device =
         size =
   block_size =
        atime =
        mtime =
        ctime =
        btime =
   hard_links =
      symlink = 0
         type =
   attributes =
volume_serial =
      file_id =

...
```